AACT-561: Remove connection information from application.rb and use database.yml instead

Javier-Jimenez-18 opened the pull request:
- Removed instances of environment variables for these database urls: AACT_BACK_DATABASE_URL, AACT_ADMIN_DATABASE_URL, and AACT_PUBLIC_DATABASE_URL.

- Replaced them with references to the database.yml

- Tested that everything still works:

**Loading Studies**

<img width="1280" alt="Screen Shot 2023-09-21 at 5 32 56 PM" src="https://github.com/ctti-clinicaltrials/aact/assets/81119399/0a1ca2fe-ac42-45ce-8c9c-664bc0601802">

**Restoring the Database**

<img width="1242" alt="Screen Shot 2023-09-21 at 5 53 54 PM" src="https://github.com/ctti-clinicaltrials/aact/assets/81119399/b3fb2a47-adbf-44a7-9179-5056e262cf83">

<img width="1163" alt="Screen Shot 2023-09-21 at 5 58 19 PM" src="https://github.com/ctti-clinicaltrials/aact/assets/81119399/917e6d51-f238-426c-9443-23ce6e99c34d">

kostik700015 reviewed (approved):
Tests passed, DB restore, LGTM!

Javier-Jimenez-18 commented:
- Removed database connection variables in application.rb file for AACT back, admin, public, and hostname for test and non-test rails environments.
- Removed unused database connection variables from .env.example, README.md, and spec file. 
- Corrected wording, spelling, and layout in README.md file.
- Tested that everything still works:

Loading Studies

<img width="1280" alt="Screen Shot 2023-09-27 at 6 08 35 PM" src="https://github.com/ctti-clinicaltrials/aact/assets/81119399/1ea3f288-2b29-4956-942e-48da96981ca7">


<img width="1280" alt="Screen Shot 2023-09-27 at 6 08 55 PM" src="https://github.com/ctti-clinicaltrials/aact/assets/81119399/2085d93e-e30b-47ac-91b5-61cc681985ac">

Restoring the Database

<img width="1280" alt="Screen Shot 2023-09-27 at 6 09 16 PM" src="https://github.com/ctti-clinicaltrials/aact/assets/81119399/4ea12a44-cdc2-4b4d-8016-d991d7443358">

micronix commented:
@Javier-Jimenez-18 some of the tests were failing, you didn't notice because right now not all of our tests are passing on aact-core, but take a look at how I fixed it in the last PR. I think this looks good now.
